diffrax/_brownian/tree.py

```diff
@@ -474,22 +465,17 @@ def _brownian_arch(
             bhh_st_tu = (bhh_st, bhh_tu)
 
             w_t = w_s + w_st
-            w_stu = (w_s, w_t, w_u)
 
             bhh_t = bhh_s + bhh_st + 0.5 * (t * w_s - s * w_t)
             bhh_stu = (bhh_s, bhh_t, bhh_u)
-            bkk_stu = None
-            bkk_st_tu = None
 
         else:
-            assert _state.bhh_s_u_su is None
```

Comment:
I think we lost this assert statement in the refactor.

I'm a little antsy about the refactor here -- easiser to be sure this is doing the desired thing (e.g. like the above assert statement) if it's just the `if` statement that changes to accomodate hew new syntax.

By the way, note that the original version also used single-assignment, e.g. `bhh_stu` was assigned to precisely once, regardless of which branch was taken. In this version, it's instead assigned to as `None` at the start, and then overriden inside one of the `if` branches. Generally speaking single-assignment tends to be (I believe!) a more reliable programming style, as you don't rely on default values. (This has been a staple of FP languages for a while; more recently single-assignment is something that has been popularised by Rust.)

Reply:
Understood. I have reintroduced the assertion and ensured single assignment. If you would like me to revert all but the changes to the if statement then no worries. I couldn't find one a quick Look, but it would be nice if ruff/pyright had an option to flag multi-assignment?

Reply:
I think changing just the if statement is probably what I'd go for here. I'm not worried about small amounts of code duplication between the branches of the if statement (if anything I think it's easier to read when you don't have to worry about keeping the in-common code correct between both branches), and I think this will be easier for to review!